stackstorm-nexus3/actions/lib/base.py

```diff
@@ -0,0 +1,178 @@
+import sys
+from exception import *
+from nexuscli.repository import Repository
+from nexuscli.nexus_client import NexusClient
+from nexuscli.exception import *
+
+from st2common.runners.base_action import Action
+
+__all__ = [
+    'BaseAction'
+]
+
+REQUIRED_FIELDS = ['url', 'user', 'password']
+
+
+class BaseAction(Action):
+
+    def __init__(self, config):
+        super(BaseAction, self).__init__(config)
+        self.nexus_profiles = self.config.get('profiles', {})
+        self.nexus_client = None
+        self.dial_config = None
+
+    def validate_config(self, c_profile):
+        """ Validate connection configuration
+
```

Comment:
Don't need this blank line.

Reply:
You may want to add a bit more information to the docstring for this function. Mention why you aren't raising exceptions in this function:
```
This function returns a tuple of (is_valid, errors), where errors is
a list of all of the missing fields in a profile. This is avoid
making users having to correct each missing field at a time.
```

or something like that.